app/controllers/declaration/BorderTransportController.scala

```diff
@@ -54,10 +58,15 @@ class BorderTransportController @Inject()(
       .fold(
         (formWithErrors: Form[BorderTransport]) => Future.successful(BadRequest(border_transport(formWithErrors))),
         borderTransport =>
-          customsCacheService
-            .cache[BorderTransport](cacheId, BorderTransport.formId, borderTransport)
-            .map(_ => Redirect(TransportDetailsController.displayForm()))
+          for {
+            _ <- updateCache(journeySessionId, borderTransport)
+            _ <- customsCacheService.cache[BorderTransport](cacheId, BorderTransport.formId, borderTransport)
+          } yield Redirect(TransportDetailsController.displayForm())
       )
   }
 
+  private def updateCache(sessionId: String, formData: BorderTransport): Future[Either[String, ExportsCacheModel]] =
+    updateHeaderLevelCache(sessionId, model => {
```

Comment:
I see this method or similar a lot, is there any value in pulling it out somehow?

Reply:
> I see this style of method or similar a lot, is there any value in pulling it out somehow?
I did not understand what do you mean? Are you talking about the `updateCache` method?

Reply:
Discussed with Paulo, we can leave it for now